wordpress/user.py

```diff
@@ -0,0 +1,41 @@
+import wordpress
+from wordpress.client import APIClient
+
+
+class WPUser:
+    email = None
+    roles = None
+    user_id = None
+    
+    def __init__(self, client=APIClient(), logger=wordpress.logger):
+        self.logger = logger
+        self.roles = set()
+        self.client = client
+
+
+    def fetch(self, email):
+        response = self.client.get_request(f'users/?search={email}')
+        body = response.json()
+        if len(body):
+            user_id = body[0]['id']
+            response = self.client.post_request(f'users/{user_id}')
```

Comment:
really? WP makes us issue a POST request to get a user? :see_no_evil:

Reply:
No, sorry, same effect with a GET... well catched, i copied the post here since it's needed to update the roles... and I saw that an empty post also was getting the user.

Reply:
Now testing I remember why I discarded GET... it does not return the 'roles' attribute of the user. I will investigate a bit more if this can be easily get with a get and some parameters to avoid using empty POST.

Reply:
Discovered that even if the WP api allows multiple roles for a user: 
![image](https://user-images.githubusercontent.com/56868560/119857481-e445de00-bf13-11eb-977c-17b9be4d6039.png)

The Wordpress admin panel user editor does not, and this may create management problems since it won't show if the user has a partner role and if saved again using the wordpress admin it may remove it.
![image](https://user-images.githubusercontent.com/56868560/119857203-a9dc4100-bf13-11eb-9a7b-e3545476b639.png)

Maybe we should just make sure that the current role is "customer" (to not replace an "admin" role for example), and replace "customer" which is the role used by woocomerce newly created users.